src/helper/git.ts

```diff
@@ -1,47 +1,47 @@
-import { execSync } from 'child_process';
-import path from 'path';
-import rimraf from 'rimraf';
+import * as deps from '../deps.deno.ts';
 
-function IsInGitRepository(root: string): boolean {
+async function IsInGitRepository(root: string) {
   try {
-    execSync('git rev-parse --is-inside-work-tree', {
+    await deps.exec({
+      cmd: 'git rev-parse --is-inside-work-tree',
       cwd: root,
-      stdio: 'ignore',
     });
     return true;
   // eslint-disable-next-line no-empty
   } catch (_) {}
   return false;
 }
 
-function IsInMercurialRepository(root: string): boolean {
+async function IsInMercurialRepository(root: string) {
   try {
-    execSync('hg --cwd . root', { cwd: root, stdio: 'ignore' });
+    deps.exec({ cwd: root, cmd: 'hg --cwd . root' });
```

Comment:
Interesting to support mercurial

Reply:
Thats sounds like feature request for another PR.

And, first of i wan't to some users, feels like i won't do that if noone uses mercurial.

Reply:
I never met anyone who is using Mercurial, and I'd put this to the very bottom of priorities. I just saw this code and was surprised you had thought of the tool at all.